### ISSUE TITLE
[BUG] Problem after upgrade

### ISSUE THREAD

**Issue by ghost** [SparebankenVest/azure-key-vault-to-kubernetes#47]
**Describe the bug**
After updating to v1.0.2 the controller shows in the log: 
`time="2020-03-18T11:44:23Z" level=error msg="error syncing 'default/tls-cluster-wildcard-certificate': AzureKeyVaultSecret.spv.no \"tls-cluster-wildcard-certificate\" is invalid: apiVersion: Invalid value: \"spv.no/v1\": must be spv.no/v1alpha1, requeuing"`

**To Reproduce**
Upgrade from 0.0.x to 1.0.2 with already created AzureKeyVaultSecrets in the Cluster

**Expected behavior**
The synch should work without complain

**Logs**
```
time="2020-03-18T11:32:22Z" level=info msg="Log level set to 'info'"
W0318 11:32:22.799495       1 client_config.go:551] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2020-03-18T11:32:22Z" level=info msg="Creating event broadcaster"
time="2020-03-18T11:32:22Z" level=info msg="Setting up event handlers"
time="2020-03-18T11:32:22Z" level=info msg="Starting AzureKeyVaultSecret controller"
time="2020-03-18T11:32:22Z" level=info msg="Waiting for informer caches to sync"
time="2020-03-18T11:32:22Z" level=info msg="Starting workers"
time="2020-03-18T11:32:22Z" level=info msg="Started workers"
time="2020-03-18T11:32:22Z" level=info msg="Successfully synced AzureKeyVaultSecret 'default/tls-cluster-wildcard-certificate' with Kubernetes Secret"
time="2020-03-18T11:32:22Z" level=info msg="Successfully synced AzureKeyVaultSecret 'default/tls-cluster-wildcard-certificate' with Kubernetes Secret"
time="2020-03-18T11:33:22Z" level=info msg="Exporting certificate with private key: true"
time="2020-03-18T11:33:23Z" level=error msg="error syncing 'default/tls-cluster-wildcard-certificate': AzureKeyVaultSecret.spv.no \"tls-cluster-wildcard-certificate\" is invalid: apiVersion: Invalid value: \"spv.no/v1\": must be spv.no/v1alpha1, requeuing"
time="2020-03-18T11:34:22Z" level=info msg="Exporting certificate with private key: true"
time="2020-03-18T11:34:23Z" level=error msg="error syncing 'default/tls-cluster-wildcard-certificate': AzureKeyVaultSecret.spv.no \"tls-cluster-wildcard-certificate\" is invalid: apiVersion: Invalid value: \"spv.no/v1\": must be spv.no/v1alpha1, requeuing"
time="2020-03-18T11:44:22Z" level=info msg="Exporting certificate with private key: true"
time="2020-03-18T11:44:23Z" level=error msg="error syncing 'default/tls-cluster-wildcard-certificate': AzureKeyVaultSecret.spv.no \"tls-cluster-wildcard-certificate\" is invalid: apiVersion: Invalid value: \"spv.no/v1\": must be spv.no/v1alpha1, requeuing"
```

Helm logs: 
```
helm3 ls
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                                   APP VERSION
akv-controller  env-injector    8               2020-03-18 11:30:40.563095938 +0000 UTC deployed        azure-key-vault-controller-1.0.2        1.0.2
akv-injector    env-injector    8               2020-03-18 11:30:35.355556568 +0000 UTC deployed        azure-key-vault-env-injector-1.0.2      1.0.2
```

**Comment by Blueye1993** [comment]
I was able to fi the problem by deleting all akvs from the cluster and recreate them with the same settings as before. Now in the logs I see: 

```
time="2020-03-18T12:03:22Z" level=info msg="Exporting certificate with private key: true"
time="2020-03-18T12:03:23Z" level=info msg="Successfully synced AzureKeyVaultSecret 'default/tls-cluster-wildcard-certificate' with Azure Key Vault"
time="2020-03-18T12:03:23Z" level=info msg="Successfully synced AzureKeyVaultSecret 'default/tls-cluster-wildcard-certificate' with Kubernetes Secret"
```

**Comment by torresdal** [comment]
@Blueye1993 Do you know the exact version you upgraded from?

**Comment by torresdal** [comment]
At some point we tried to upgrade the api-version from v1alpha1 to v1 (between image versions >=1.0.0-beta.4<=1.0.0-beta.8. Turned out not to be a trivial task, so we reverted back. Versions >= 1.0.0-beta.9 should be OK. Unless you were running any of these Beta image versions, I can't see what could cause this.

During this time, a new Helm chart was NOT released.

**Comment by Blueye1993** [comment]
we came from `0.1.15`

**Comment by torresdal** [comment]
Ok. Strange. Hopefully we can recreate and see what’s going on. Thanks for reporting and providing the workaround.

**Comment by Blueye1993** [comment]
Thanks for the quick reply

**Comment by Blueye1993** [comment]
Ohh here are the helm values.yamls I used for instalation: 
env injector
```
installCrd: false
envImage:
  repository: ${ACR_NAME}.azurecr.io/spvest/azure-keyvault-env
  tag: 1.0.2
image:
  repository: ${ACR_NAME}.azurecr.io/spvest/azure-keyvault-webhook
  tag: 1.0.2

```
controller 
```
image:
  repository: ${ACR_NAME}.azurecr.io/spvest/azure-keyvault-controller
  tag: 1.0.2
```
sidenote: The images were imported from docker hub without any modification

**Comment by torresdal** [comment]
@Blueye1993 problem found and fixed. Fix will be available in the next release (v 1.1.0).

**Comment by ChaosGramer** [comment]
Awesome that was fast
